src/pages/registration/FilesAndLicensePanel.tsx

```diff
@@ -412,94 +141,122 @@ export const FilesAndLicensePanel = ({ uppy }: FilesAndLicensePanelProps) => {
-                <DoiField canEditDoi={canEditFiles} />
-              ) : (
-                <TextField
-                  fullWidth
-                  variant="filled"
-                  label={t('registration.files_and_license.link_to_resource')}
-                  disabled={!canEditFiles}
-                  value={
-                    associatedLinkIndex >= 0 ? (associatedArtifacts[associatedLinkIndex] as AssociatedLink).id : ''
-                  }
-                  error={associatedLinkHasError}
-                  helperText={
-                    associatedLinkHasError
-                      ? (errors.associatedArtifacts?.[associatedLinkIndex] as FormikErrors<AssociatedLink>).id
-                      : null
-                  }
-                  data-testid={dataTestId.registrationWizard.files.linkToResourceField}
-                  onChange={(event) => {
-                    const inputValue = event.target.value;
-                    if (inputValue) {
-                      if (associatedLinkIndex < 0) {
-                        const newAssociatedLink: AssociatedLink = {
-                          type: 'AssociatedLink',
-                          id: inputValue,
-                        };
-                        push(newAssociatedLink);
-                        const nullAssociatedArtifactIndex = associatedArtifacts.findIndex(
-                          associatedArtifactIsNullArtifact
-                        );
-                        if (nullAssociatedArtifactIndex > -1) {
-                          remove(nullAssociatedArtifactIndex);
-                        }
-                      } else {
-                        const fieldName = `${name}[${associatedLinkIndex}].${SpecificLinkFieldNames.Id}`;
-                        setFieldValue(fieldName, inputValue);
-                        setFieldTouched(fieldName);
-                      }
-                    } else {
-                      const associatedArtifactsBeforeRemoval = associatedArtifacts.length;
-                      remove(associatedLinkIndex);
-                      if (associatedArtifactsBeforeRemoval === 1) {
-                        // Ensure field is set to touched even if it's empty
-                        setFieldTouched(name);
-                      }
-                    }
-                  }}
+              {filesToPublish.length > 0 && (
+                <FileList
+                  title={t('registration.files_and_license.files_to_publish')}
+                  files={filesToPublish}
+                  uppy={uppy}
+                  remove={remove}
+                  name={name}
+                />
+              )}
+              {filesNotToPublish.length > 0 && (
+                <FileList
+                  title={t('registration.files_and_license.files_are_not_published')}
+                  files={filesNotToPublish}
+                  uppy={uppy}
+                  remove={remove}
+                  name={name}
+                  archived
```

Comment:
Ser det er en god del av `FileList` som bare er relevant når `archived=true` og motsatt, så det ligger en del funksjonalitet der som ikke er relevant uansett om archieved er true eller false (gjelder også de to andre `FileList` her). Når vi viser arkiverte filer vet vi at vi aldri trenger alskens logikk for embargo, RRS, fil-versjon, lisens, osv, men nå vil alt alltid importeres for alle disse 3 listene selv om det ikke trengs.

Det koker egentlig ned til det evige spørsmålet om generalisering vs overgeneralisering, som så mye annet 😅 

Om vi hadde splitta opp `FileList` til feks `UnpublishableFileList` etc måtte vi sikkert duplisert noe sjekker for tilgangskontroll osv da, så mulig du har tenkte tanken allerede og kommet frem til at det var best sånn? I så fall helt fair

Reply:
Som jeg skriver i teksten så fantes `UnpublishableFileList` allerede, men jeg tok den bort fordi det var veldig mye kode som var lik. Jeg er jo vanligvis den som jobber for å _dele opp_ filer for å unngå veldig mye if/else-logikk, men tanken min her var at filene var veldig like, og logikken begrenset seg til å ta bort kolonner. Det er vanskelig å vite om dette vil stemme med endringene som kommer når jeg skal re-designe tabellene, men en stor grunn til å gjøre det slik er fordi det da blir lettere å aligne tabellene slik at kolonne-grensene går på samme sted, som du ser i designet:

![image](https://github.com/BIBSYSDEV/NVA-Frontend/assets/1130244/716e7309-435d-4c0b-9a96-ef8deb41d4b0)

I motsetning til slik det er i dag (også i denne PR'en):

![image](https://github.com/BIBSYSDEV/NVA-Frontend/assets/1130244/d97d2d39-5f14-4334-8805-5bcc01119992)

Reply:
Mange plasser har vi brukt prosent for å sette bredde, som også kan hjelpe med å aligne, feks type sånn:
```
<TableCell width="40%">Fil</TableCell>
<TableCell width="20%">Marker for publisering</TableCell>
<TableCell width="20%">Versjon</TableCell>
<TableCell width="20%">Lisens</TableCell>
```
Om vi bare skal ha to kolonner kan vi aligne med
```
<TableCell width="40%">Fil</TableCell>
<TableCell width="60%">Marker for publisering</TableCell>
```

Men helt greit at det er i en fil som det er nå altså!